packages/services/cdn-worker/src/artifact-handler.ts

```diff
@@ -0,0 +1,59 @@
+import type { KeyValidator } from './key-validation';
+import { Response } from '@whatwg-node/fetch';
+import itty from 'itty-router';
+import zod from 'zod';
+import { InvalidAuthKeyResponse, MissingAuthKeyResponse } from './errors';
+
+type ArtifactRequestHandler = {
+  getArtifactUrl: (
+    targetId: string,
+    artifactType: 'metadata' | 'sdl' | 'services',
+  ) => Promise<string | null>;
+  isKeyValid: KeyValidator;
+};
+
+const ParamsModel = zod.object({
+  targetId: zod.string(),
+  artifactType: zod.union([zod.literal('metadata'), zod.literal('sdl'), zod.literal('services')]),
+});
+
+const authHeaderName = 'x-hive-cdn-key' as const;
+
+export const createArtifactRequestHandler = (deps: ArtifactRequestHandler) => {
+  const router = itty.Router();
```

Comment:
```suggestion
  const router = itty.Router<Request & itty.Request>();
```
I think then you can remove the casting below.

Reply:
![image](https://user-images.githubusercontent.com/14338007/203992158-c2004843-eb39-4418-b5e8-6aa6d1c8da2d.png)

Reply:
Solved in https://github.com/kamilkisiela/graphql-hive/pull/692/commits/4b292297f5807bb581dcbf54abf2c0474768d291